nautobot/extras/models/metadata.py

```diff
@@ -135,3 +141,249 @@ def save(self, *args, **kwargs):
+
+                # Validate JSON
+                elif metadata_type_data_type == MetadataTypeDataTypeChoices.TYPE_JSON:
+                    pass
+                    # TODO uncomment this when MetaDataType validation_minimum, validation_maximum and validation_regex fields are implemented
+                    # if (
+                    #     self.metadata_type.validation_regex
+                    #     or self.metadata_type.validation_minimum is not None
+                    #     or self.metadata_type.validation_maximum is not None
+                    # ):
+                    #     json_value = json.dumps(value)
+                    #     if (
+                    #         self.metadata_type.validation_minimum is not None
+                    #         and len(json_value) < self.metadata_type.validation_minimum
+                    #     ):
+                    #         raise ValidationError(
+                    #             f"Value must be at least {self.metadata_type.validation_minimum} characters in length"
+                    #         )
+                    #     if (
+                    #         self.metadata_type.validation_maximum is not None
+                    #         and len(json_value) > self.metadata_type.validation_maximum
+                    #     ):
+                    #         raise ValidationError(
+                    #             f"Value must not exceed {self.metadata_type.validation_maximum} characters in length"
+                    #         )
+                    #     if self.metadata_type.validation_regex and not re.search(
+                    #         self.metadata_type.validation_regex, json_value
+                    #     ):
+                    #         raise ValidationError(f"Value must match regex '{self.metadata_type.validation_regex}'")
+
+                # Validate integer
+                elif metadata_type_data_type == MetadataTypeDataTypeChoices.TYPE_INTEGER:
+                    try:
+                        value = int(value)
+                    except ValueError:
+                        raise ValidationError("Value must be an integer.")
+                    # TODO uncomment this when MetaDataType validation_minimum, validation_maximum and validation_regex fields are implemented
+                    # if self.metadata_type.validation_minimum is not None and value < self.metadata_type.validation_minimum:
+                    #     raise ValidationError(f"Value must be at least {self.metadata_type.validation_minimum}")
+                    # if self.metadata_type.validation_maximum is not None and value > self.metadata_type.validation_maximum:
+                    #     raise ValidationError(f"Value must not exceed {self.metadata_type.validation_maximum}")
+                # Validate integer
+                elif metadata_type_data_type == MetadataTypeDataTypeChoices.TYPE_FLOAT:
+                    try:
+                        value = float(value)
```

Comment:
Do we ever update `self._value` with the type-coerced new `value`? I'm not seeing it but it feels like it should be necessary?